llvmlibc-samples/src/llvmlibc/baremetal-semihosting/hello.c

```diff
@@ -19,7 +19,13 @@
 #include <string.h>
 #include <math.h>
 
-/* Example that uses heap, string and math library */
+// Implementation of errno
+int *__llvm_libc_errno() {
+  static int errno;
```

Comment:
Sorry to do another pedantic one, but I recommend calling that internal variable something other than `errno`. If a user starts from this code and adds `#include <errno.h>`, then the name `errno` will surely become a macro, making this function into nonsense. Any other name is fine.

Reply:
Thanks for spotting. I've used a name that doesn't contain errno.